components/server/src/server.rs

```diff
@@ -334,6 +336,7 @@ impl<ER: RaftEngine> TiKvServer<ER> {
             flow_info_receiver: None,
             sst_worker: None,
             quota_limiter,
+            causal_ts_provider: None,
```

Comment:
Why add causal_ts_provider into TiKvServer?

Reply:
We need pass `causal_ts_provider` to `cdc::Endpoint` and `backup::Endpoint`. `cdc::Endpoint` is created later in this function, but cannot pass an Optional variable here as lifecyle compile issue and `backup::Endpoint` is created in another function. So store `causal_ts_provider` in TiKVServer is the easiest and reasonable way I found.